### PR TITLE
Add Command Line Interface and Update README

### PR DESCRIPTION
**Pull request by andrewDoing**
https://github.com/andrewDoing/hydrate/issues/4
https://github.com/andrewDoing/hydrate/issues/6

Added barebones CLI and updated the docs to reflect it.

Ran through Pyflakes, pycodestyle, and pydocstyle linters.

### CODE REVIEW

**Comment by mtarng** [comment]
@andrewDoing what about adding a flag to specify a `kubeconfig` file. Could be required or optional but it gives the tool and user a bit more flexibility.

File: hydrate.py [hunk 1]
```diff
@@ -34,6 +36,36 @@ def get_deployments_for_all_namespaces():
...
+    parser = argparse.ArgumentParser(
+        description='Generate a component.yaml file for your cluster.')
+    parser.add_argument(
+        'run',
+        help='Generate component.yaml for current configuration.')
+    parser.add_argument(
+        '-n', '--name',
+        action='store',
+        default='hydrated-cluster',
+        help='Specify the name of the main component.')
```
Comment (edaena) [inline hydrate.py]: Another way of putting the help can be "Name of the main component (default: hydrated-cluster)"

File: hydrate.py [hunk 2]
```diff
@@ -34,6 +36,36 @@ def get_deployments_for_all_namespaces():
...
+    parser.add_argument(
+        '-o', '--output',
+        action='store',
+        default=os.getcwd(),
+        help='Specify path of the generated component.yaml.',
```
Comment (edaena) [inline hydrate.py]: "Output path for the generated component.yaml"

File: hydrate.py [hunk 3]
```diff
@@ -34,6 +36,36 @@ def get_deployments_for_all_namespaces():
...
+        metavar='PATH')
+    parser.add_argument(
+        '-v', '--verbose',
+        action='store_true',
+        help='Print more output.')
```
Comment (edaena) [inline hydrate.py]: "Verbose output logs."

File: hydrate.py (hunk 1 above)
Comment (andrewDoing) [reply hydrate.py]: Added

File: hydrate.py (hunk 2 above)
Comment (andrewDoing) [reply hydrate.py]: Added

File: hydrate.py (hunk 3 above)
Comment (andrewDoing) [reply hydrate.py]: Added

**Comment by andrewDoing** [comment]
OK now it should be ready for a final review.

**Review by mtarng — APPROVED** [review]

**Pull request merged**